mempool/mempool.go

```diff
@@ -81,8 +81,11 @@ type Mempool struct {
 	recheckEnd           *clist.CElement // re-checking stops here
 	notifiedTxsAvailable bool
 	txsAvailable         chan struct{} // fires once for each height, when the mempool is not empty
-	// Filter mempool to only accept txs for which filter(tx) returns true.
-	filter func(types.Tx) bool
+	// Filter mempool to only accept txs for which preCheckFilter(tx) returns true.
+	// This is called before each check tx
+	preCheckFilter func(types.Tx) bool
+	// Filter mempool to only accept txs for which postCheckFilter(tx) returns true.
+	postCheckFilter func(types.Tx, *abci.ResponseCheckTx) bool
```

Comment:
These should be typed a la: 

``` go
type preCheckFilterFunc func(types.Tx) bool

type postCheckFilterFunc func(types.Tx, *abci.ResponseCheckTx) bool
```

Likely also made public if the option functions are called from outside the mempool with it.